evaluation/tasks/piqa/piqa.py

```diff
@@ -0,0 +1,83 @@
+# Module for any additional processing required for the TyDi QA dataset
+# HuggingFace dataset link: https://huggingface.co/datasets/piqa
+from datasets import load_dataset
+from jinja2 import Template
+from torch.utils.data import Dataset
+from tqdm import tqdm
+
+from evaluation.tasks.auto_task import AutoTask
+
+
+TEMPLATE = Template(
+    """
+Given a goal and 2 solutions, choose the most appropriate solution.
+Goal: {{goal}}
+{{'Solution 1'}}: {{sol1}}
+{{'Solution 2'}}: {{sol2}}
+Answer:
+    """
+)
+
+
+class PIQADataset(Dataset):
+    def __init__(self, tokenizer):
+        super().__init__()
+        piqa = load_dataset("piqa", split="validation")
+        self.items = []
+
+        for sample in piqa:
+            prompt = TEMPLATE.render(
+                goal=sample["goal"],
+                sol1=sample["sol1"],
+                sol2=sample["sol2"],
+            )
+
+            # Tokenize and construct this sample
+            inputs = tokenizer(
+                prompt,
+                return_tensors="pt",
+            )
+            self.items.append(
+                {
+                    "prompt": prompt,
+                    "input_ids": inputs["input_ids"],
+                    "attention_mask": inputs["attention_mask"],
+                    "input_len": inputs["attention_mask"].shape[1],
+                    "label": [sample["sol1"], sample["sol2"]][sample["label"]],
+                }
+            )
+
+    def __len__(self):
+        return len(self.items)
+
+    def __getitem__(self, index):
+        return self.items[index]
+
+
+class PIQATask(AutoTask):
+    @staticmethod
+    def get_display_name() -> str:
+        return "piqa"
+
+    def evaluate(self) -> None:
+        dataset = PIQADataset(self.tokenizer)
+
+        substring_matches = 0
+        for sample in tqdm(dataset):
```

Comment:
In `tydiqa_secondary.py`, we have 

```python
tqdm(dataset, desc=f"Evaluating {self.get_display_name()}"):
```

So I suggest to stay consistent with the `tqdm` signature. (I should fix this in the WMT PR as well). Or better, would there be a way to somehow abstract this portion out in the `AutoTask` class so that we don't have to worry about uniformity across subclasses? @wilsonyhlee

Reply:
Good point. I'd fix this right away. If there's no way to abstract this line, we could have a `template_task.py` that emphasizes these format (e.g. consistency with `tqdm` signature, use of `self.metric`, etc.).

Reply:
Hmm I wouldn't try to abstract this portion in the parent class. It would get very messy. I agree with Tian that a template_task (that we have to create for issue #39 and #40 anyway) will be sufficient.

Reply:
The template approach sounds good to me!